lib/std/os.zig

```diff
@@ -4354,6 +4354,35 @@ pub fn fork() ForkError!pid_t {
     }
 }
 
+pub fn getPageSize() usize {
+    switch (builtin.cpu.arch) { // archs with fixed size page size
+        .wasm32, .wasm64 => 64 * 1024,
+        else => {},
+    }
+
+    switch (builtin.os.tag) { // kernel chooses the page size
+        .linux => {
+            if (builtin.link_libc) {
+                return std.c.sysconf(std.c._SC.PAGESIZE);
+            } else {
+                return linux.getauxval(std.elf.AT_PAGESZ);
+            }
+        },
+        .windows => {
+            var si: windows.SYSTEM_INFO = undefined;
+            windows.kernel32.GetSystemInfo(&si);
```

Comment:
Might want to use `NtQuerySystemInformation(SystemBasicInformation, ...)` instead given #1840 (despite Microsoft's explicit recommendation not to).

Reference: https://www.geoffchappell.com/studies/windows/km/ntoskrnl/inc/api/ntexapi/system_basic_information.htm

Reply:
1. It is also used in lib/std/process.zig:1197
```zig
        .windows => {
            var sbi: std.os.windows.SYSTEM_BASIC_INFORMATION = undefined;
            const rc = std.os.windows.ntdll.NtQuerySystemInformation(
                .SystemBasicInformation,
                &sbi,
                @sizeOf(std.os.windows.SYSTEM_BASIC_INFORMATION),
                null,
            );
            if (rc != .SUCCESS) {
                return error.UnknownTotalSystemMemory;
            }
            return @as(usize, sbi.NumberOfPhysicalPages) * sbi.PageSize;
        },
```
2. For this simple use case I think this is ok. However, more complex stuff definitely not: 
```
"Microsoft broke backwards compatibility and GetSystemInfo, GetThreadTimes, NtQuerySystemInformation and quite a few other functions are busted and broken on Windows 11 when you have more than 1 processor group.

Starting with Windows 11, process and thread affinities span all processors in the system, across all processor groups by default. Each processor group has completely different processor configuration and statistics... GetSystemInfo, GetThreadTimes and NtQuerySystemInformation only return values specific to the current thread' processor group affinity and they don't query multiple processor groups.

The Windows 11 kernel scheduler will execute your process/thread on different processor groups while you're calling those functions and this causes those functions to return completely different set of values that cannot be compared. Every program using those functions currently shows incorrect information exactly like what you've mentioned in your post.

You either need to call SetThreadGroupAffinity and restrict the thread to a specific group so those functions don't switch group and return incorrect values (and ignore statistics for other processors) or update your code and query the statistics for all processor groups by calling NtQuerySystemInformationEx."
```
from https://learn.microsoft.com/en-us/answers/questions/1183193/totalprocessortime-differs-between-windows-10-and.

Reply:
Added. Almost a copy-pasta.